### PR TITLE
user story for creating a new booking

### PR DESCRIPTION
**Pull request by joshwbarnes**
Hi guys,

Here is the user story for creating a new booking without the view. I'm getting on with the view now..

### CODE REVIEW

File: app/models/booking.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Booking < ApplicationRecord
+  belongs_to :user
+  belongs_to :monster
```
Comment (Aya1368) [inline app/models/booking.rb:3]: booking has many monster, monster belongs to booking
also, I think we need parameter for total_price?

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+class BookingsController < ApplicationController
+  def create
```
Comment (Aya1368) [inline app/controllers/bookings_controller.rb:2]: missing def new

File: app/controllers/bookings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  # def destroy
+  #   make_booking
+  #   @bookig.destroy
+  #   redirect_to root_path
+  # end
+
+  private
+
+  def booking_params
+    params.require(:booking).permit(:start_date, :end_date, :total_price)
```
Comment (lucieroland) [inline app/controllers/bookings_controller.rb:38]: missing monster_id and user_id as parameters in the `.permit()`?

File: app/controllers/bookings_controller.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
+class BookingsController < ApplicationController
+  def create
+    @monster = Monster.find(params[:monster_id])
+    @booking = Booking.new(booking_params)
+    @booking.monster = @monster
+    @booking.user = current_user
```
Comment (lucieroland) [inline app/controllers/bookings_controller.rb:6]: do we need that line?

File: app/controllers/bookings_controller.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    if @booking.save
+      redirect_to booking_path(@booking)
+    else
+      redirect_to monster_path(@monster)
```
Comment (lucieroland) [inline app/controllers/bookings_controller.rb:10]: could also be 'else render' depending if we want to redirect to the monster page or to the booking form

File: app/controllers/bookings_controller.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
+class BookingsController < ApplicationController
+  def create
+    @monster = Monster.find(params[:monster_id])
+    @booking = Booking.new(booking_params)
+    @booking.monster = @monster
+    @booking.user = current_user
+    if @booking.save
+      redirect_to booking_path(@booking)
```
Comment (Aya1368) [inline app/controllers/bookings_controller.rb:8]: I think we don't have route for booking yet

File: app/controllers/bookings_controller.rb (hunk 4 above)
Comment (joshwbarnes) [reply app/controllers/bookings_controller.rb:6]: which line are you referring to?

File: app/controllers/bookings_controller.rb (hunk 6 above)
Comment (joshwbarnes) [reply app/controllers/bookings_controller.rb:8]: just added

File: app/models/booking.rb (hunk 1 above)
Comment (glenntippett) [reply app/models/booking.rb:3]: add: has_many :reviews

**Review by joshwbarnes — COMMENTED** [review]
updates we just went through together

File: config/routes.rb [hunk 7]
```diff
@@ -4,6 +4,8 @@
 
   devise_for :users
 
-  resources :monsters
+  resources :monsters do
+  resources :booking, only: [:new, :update, :destroy]
```
Comment (glenntippett) [inline config/routes.rb:8]: index this nested route

**Pull request merged**